feat(json): Add `extensions` option to load JSON files with custom extensions

altinokdarici opened the pull request:
<!--
  ⚡️ katchow! We ❤️ Pull Requests!

  If you remove or skip this template, you'll make the 🐼 sad and the mighty god
  of Github will appear and pile-drive the close button from a great height
  while making animal noises.

  Pull Request Requirements:
  * Please include tests to illustrate the problem this PR resolves.
  * Please lint your changes by running `npm run lint` before creating a PR.
  * Please update the documentation in `/docs` where necessary

  Please place an x (no spaces - [x]) in all [ ] that apply.
-->

<!-- the plugin(s) this PR is for -->

## Rollup Plugin Name: `json`

This PR contains:

- [ ] bugfix
- [x] feature
- [ ] refactor
- [ ] documentation
- [ ] other

Are tests included?

- [x] yes (_bugfixes and features will not be merged without tests_)
- [ ] no

Breaking Changes?

- [ ] yes (_breaking changes will not be merged unless absolutely necessary_)
- [x] no

If yes, then include "BREAKING CHANGES:" in the first commit message body, followed by a description of what is breaking. 

List any relevant issue numbers:

<!--
If this PR resolves any issues, list them as

  resolves #1234

where 1234 is the issue number. This will help us with house-keeping as Github will automatically add a note to those issues stating that a potential fix exists. Once the PR is merged, Github will automatically close those issues.

If an issue is only solved partially or is relevant in some other way, just list the number without "resolves".
-->

### Description

<!--
  Please be thorough and clearly explain the problem being solved.
  * If this PR adds a feature, look for previous discussion on the feature by searching the issues first.
  * Is this PR related to an issue?
-->

This PR adds support for custom file extensions in the JSON plugin, allowing handling beyond the default .json files.
Previously, the plugin only handled .json files, limiting use with custom formats (such as .resjson)

### Changes

* [`packages/json/README.md`](diffhunk://#diff-c6477b6b1ead18bdf399d3e41154c47d02e0371164593bcb41c4c9cdefb394bdR106-R111): Added documentation for the new `extensions` option that allows specifying a list of file extensions to be handled by the plugin.
* [`packages/json/src/index.js`](diffhunk://#diff-412d92e18a781647b77d5c02b40f7dd4c377fdc88c77e35013af3957135cc879R2-R16): Introduced the `extensions` option in the JSON plugin's configuration, allowing users to specify which file extensions should be processed.
* [`packages/json/test/fixtures/custom-file-extension/config.customjson`](diffhunk://#diff-505c9974c17018d995041e12c126d2bb43afd4d0969b120ad7dabeb64939f81dR1-R3): Added a test fixture with a custom file extension to verify the new feature.
* [`packages/json/test/fixtures/custom-file-extension/main.js`](diffhunk://#diff-6a9d215b0e607c6dbeb90ec58111e2dda5c1643d74e9fedc0b6d1af503877976R1-R3): Added a test case to import and validate the custom file extension handling.
* [`packages/json/test/test.js`](diffhunk://#diff-c7cb9d15a5bb164a27e222f69a3d298969d3a9f7648e8e40eddf23705be33eb3R167-R187): Added new test cases to ensure the plugin correctly processes files with custom extensions and ignores non-custom extensions.

shellscape reviewed (changes requested):
Small nits on the docs for this feature. Once those are tackled we can merge. Thank you for the PR.

altinokdarici commented on packages/json/types/index.d.ts:
```diff
@@ -33,6 +33,11 @@ export interface RollupJsonOptions {
+   * Specify the file extensions to be parsed as JSON
+   * @default ['.json']
+   */
+  extensions?: string[];
```
Done.

altinokdarici commented:
> Small nits on the docs for this feature. Once those are tackled we can merge. Thank you for the PR.

@shellscape I’ve gone ahead and addressed your comments. Let me know if there’s anything else I missed.

altinokdarici commented:
Hey @shellscape, I've addressed your comments. Can you review this PR again when you have time?

shellscape commented:
@altinokdarici looks like there's a prettier issue. please run lint and tag me again when you update

shellscape commented:
Closing as abandoned. If you find your way back to this PR please drop a comment and we'll reopen.
